Dockerfile

```diff
@@ -44,9 +48,11 @@ WORKDIR /edx/app/notes
 # Copy the requirements explicitly even though we copy everything below
 # this prevents the image cache from busting unless the dependencies have changed.
 COPY requirements/base.txt /edx/app/notes/requirements/base.txt
+COPY requirements/pip.txt /edx/app/notes/requirements/pip.txt
 
 # Dependencies are installed as root so they cannot be modified by the application user.
-RUN pip3 install -r requirements/base.txt
+RUN pip install -r requirements/pip.txt
```

Comment:
@mraarif Could you explain how pip can install pip and why we need this extra file or why we don't want to install pip the way we do in the cookiecutter? https://github.com/edx/edx-cookiecutters/blob/master/cookiecutter-django-ida/%7B%7Bcookiecutter.repo_name%7D%7D/Dockerfile#L24

Reply:
I believe cookiecutter is still using Dockerfile that installs python 3.5 and when we install `python3.8-dev` and `python3.8-venv` `pip` gets installed with them, this `pip install pip` is here to ensure a specific version of pip gets installed across services
ref: https://github.com/edx/registrar/pull/354#discussion_r497710581